routing/router.go

```diff
@@ -967,13 +968,15 @@ type LightningPayment struct {
 }
 
 // SendPayment attempts to send a payment as described within the passed
-// LightningPayment. This function is blocking and will return either: when the
+// LightningPayment. If maxFee is not nil, SendPayment will attempt to
+// find a route that requires a fee of less than *maxFee satoshis.
+// This function is blocking and will return either: when the
 // payment is successful, or all candidates routes have been attempted and
 // resulted in a failed payment. If the payment succeeds, then a non-nil Route
 // will be returned which describes the path the successful payment traversed
 // within the network to reach the destination. Additionally, the payment
 // preimage will also be returned.
-func (r *ChannelRouter) SendPayment(payment *LightningPayment) ([32]byte, *Route, error) {
+func (r *ChannelRouter) SendPayment(payment *LightningPayment, maxFee *lnwire.MilliSatoshi) ([32]byte, *Route, error) {
```

Comment:
Seems like max fee should maybe be a field on `LightningPayment` instead.

Reply:
@jimpo I totally agree, I'll make those changes now